cmd/query-tee/proxy_backend.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"context"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// ProxyBackend holds the information of a single backend.
+type ProxyBackend struct {
+	name     string
+	endpoint *url.URL
+	client   *http.Client
+	timeout  time.Duration
+
+	// Whether this is the preferred backend from which picking up
+	// the response and sending it back to the client.
+	preferred bool
+}
+
+// NewProxyBackend makes a new ProxyBackend
+func NewProxyBackend(name string, endpoint *url.URL, timeout time.Duration, preferred bool) *ProxyBackend {
+	return &ProxyBackend{
+		name:      name,
+		endpoint:  endpoint,
+		timeout:   timeout,
+		preferred: preferred,
+		client: &http.Client{
+			CheckRedirect: func(_ *http.Request, _ []*http.Request) error {
+				return errors.New("The query-tee proxy does not follow redirects.")
+			},
+			Transport: &http.Transport{
+				Proxy: http.ProxyFromEnvironment,
+				DialContext: (&net.Dialer{
+					Timeout:   30 * time.Second,
+					KeepAlive: 30 * time.Second,
+					DualStack: true,
+				}).DialContext,
+				MaxIdleConns:          10000,
+				MaxIdleConnsPerHost:   1000, // see https://github.com/golang/go/issues/13801
+				IdleConnTimeout:       90 * time.Second,
+				DisableKeepAlives:     true,
```

Comment:
Why? On one hand, we enable 10000 idle connections, on the other hand we disable idle connections completely here.

Reply:
Agree. It was non-sense. Was blindly copied from a "gateway" 🤦‍♂ . I've kept keep-alive enabled (default), which I believe for this use case makes sense.